src/WallsComputation.cpp

```diff
@@ -129,15 +129,15 @@ void WallsComputation::generateInsets(SliceLayerPart* part)
 
     constexpr float transitioning_angle = 0.5;
 
-    Polygons tubeshape = part->outline.tubeShape(line_width_0 + line_width_x * (inset_count - 1), 0).offset(-10).offset(10);
-    tubeshape.simplify(50, 50);
-    tubeshape.removeColinearEdges(0.03);
-    tubeshape.fixSelfIntersections();
-    tubeshape.removeSmallAreas(INT2MM(line_width_0/2) * INT2MM(line_width_0/2), false); // TODO: complete guess as to when arachne starts breaking, but it doesn't function well when an area is really small apearantly?
-    if (tubeshape.area() > 0)
+    Polygons prepared_outline = part->outline.offset(-10).offset(10);
+    prepared_outline.simplify(50, 50);
```

Comment:
Maybe use `constexpr` for the magic numbers in the `simplify` and `removeColinearEdges`.
These numbers will also be used by other libArachne wall generations, such as supportwalls. Maybe defined them once for the whole of CuraEngine?

Reply:
same goes btw for the division factor used by the `LimitedDistributedBeadingStrategy(... line_width_0 * 2, inset_count * 2,..)` I would like to propose to make these `constexpr` for the whole of CuraEngine

Reply:
Yeah agreed. I didn't pay much attention to this since they're likely to change (the functions themselves), but might as well make the parameters `constexpr` in case those functions _do_ stay where they are.

Reply:
Except the beading strategy parameters should maybe be up to the user ... Kostas is working on that as well.

Reply:
I've made the beading parameters user-defined:
![image](https://user-images.githubusercontent.com/19388042/90121898-d811db00-dd5c-11ea-85d4-89816a35cbbc.png)

Reply:
@konskarm looks good. This remark is unrelated to this ticket, but did you make that reusable? Such that it could also be used for support and other walls?